Author: Andreas Hoenen <andreas@hoenen-terstappen.de>
Description: Fix for BTS report #684772:
Add support for labeled chapters.
--- a/xsl/labelid.xsl
+++ b/xsl/labelid.xsl
@@ -44,12 +44,28 @@
     </xsl:choose>
   </xsl:variable>
 
+  <!-- handle labeled chapters (part 1):
+       the label replaces the chapter number,
+       and the chapter counter won't be incremented
+  -->
+  <xsl:if test="contains($rcommand,'\chapter') and @label">
+    <xsl:text>\renewcommand{\thechapter}{</xsl:text><xsl:value-of select="@label"/><xsl:text>}</xsl:text>
+  </xsl:if>
+
   <xsl:value-of select="$rcommand"/>
   <xsl:apply-templates select="$title" mode="format.title">
     <xsl:with-param name="allnum" select="$allnum"/>
   </xsl:apply-templates>
   <xsl:call-template name="label.id"/>
   <xsl:apply-templates select="title" mode="foottext"/>
+
+  <!-- handle labeled chapters (part 2)
+  -->
+  <xsl:if test="contains($rcommand,'\chapter') and @label">
+    <xsl:text>\addtocounter{chapter}{-1}</xsl:text>
+    <xsl:text>\renewcommand{\thechapter}{\arabic{chapter}}</xsl:text>
+  </xsl:if>
+
 </xsl:template>
 
 <!-- Make a section heading from a title string. It gives something like:
